src/licensedcode/cache.py

```diff
@@ -151,19 +164,35 @@ def build_index(
     Otherwise, only include the English license texts and rules (the default)
     """
     from licensedcode.index import LicenseIndex
+    from licensedcode.models import get_license_dirs
+    from licensedcode.models import get_rule_dirs
     from licensedcode.models import get_rules
+    from licensedcode.models import get_rules_from_multiple_dirs
     from licensedcode.models import get_all_spdx_key_tokens
     from licensedcode.models import get_license_tokens
     from licensedcode.models import licenses_data_dir as ldd
     from licensedcode.models import rules_data_dir as rdd
     from licensedcode.models import load_licenses
+    from licensedcode.models import load_licenses_from_multiple_dirs
     from licensedcode.legalese import common_license_words
 
     licenses_data_dir = licenses_data_dir or ldd
     rules_data_dir = rules_data_dir or rdd
 
-    licenses_db = licenses_db or load_licenses(licenses_data_dir=licenses_data_dir)
-    rules = get_rules(licenses_db=licenses_db, rules_data_dir=rules_data_dir)
+    if not licenses_db:
+        if additional_directories:
```

Comment:
@pombredanne I'm actually combining all the licenses and rules in the input directories into a single index here. In `load_licenses_from_multiple_dirs`, all the licenses (including the ones in the ScanCode database) are combined into a single license database, and the same is done with all the rules. I'll update this code with a comment so that this is more clear. # # # #